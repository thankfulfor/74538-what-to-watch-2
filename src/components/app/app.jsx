import * as React from 'react';
import PropTypes from 'prop-types';
import {WelcomeScreen} from '../welcome-screen/welcome-screen.jsx';

export const App = (props) => {
  const {films, onClick} = props;
  return (
    <WelcomeScreen films={films} onClick={onClick}/>
  );
};

App.propTypes = {
  films: PropTypes.array.isRequired,
  onClick: PropTypes.func.isRequired
};


