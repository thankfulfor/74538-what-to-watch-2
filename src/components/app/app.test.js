import React from 'react';
import renderer from 'react-test-renderer';
import {App} from './app';

const films = [{
  picture: `img/we-need-to-talk-about-kevin.jpg`,
  name: `We need to talk about Kevin`,
}];

it(`App корректно рендерится после перезапуска`, () => {
  const clickHandler = function () {};
  const tree = renderer
    .create(<App
      films={films}
      onClick={clickHandler}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
