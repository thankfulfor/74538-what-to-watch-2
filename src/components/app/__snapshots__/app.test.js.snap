// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App корректно рендерится после перезапуска 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(SignIn)",
        "type": [Function],
      }
    }
    exact={true}
    path="/login"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(MyList)",
        "type": [Function],
      }
    }
    exact={true}
    path="/mylist"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(MoviePage)",
        "type": [Function],
      }
    }
    exact={true}
    path="/films/:id"
  />
</Switch>
`;
